File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
   int count = 0;
    public int solution(int n) { 
        if(n==0) return 0;
        if(n==1) return 1;
            n = solution(1,0,n);
            return n%1234567;
    }

    public int solution(int sum,int num2, int number) {    
        count++;
        int sum2 = (sum + num2) % 1234567;        
        sum = num2;        
        if(count == number){
            count = 0;
            return sum2;
        }        
        return solution(sum,sum2,number);
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n){
    let arr = [];
    arr.push(0);
    arr.push(1);
    for (let i = 2; i <= n; i++) arr.push(arr[i - 1] + arr[i - 2] % 1234567);
    return arr[n] % 1234567;
}